packages/serve/index.ts

```diff
@@ -22,30 +22,41 @@ import { processPromise } from "@webpack-cli/utils/resolve-packages";
  * @returns {Void}
  */
 
-interface ConfigType {
-	installCmd: string;
-	dependency: string;
-	devDependency: string;
-	optionalDependency: string;
+interface Commands {
+	dependency: string[];
+	devDependency: string[];
+	optionalDependency: string[];
 }
 
-const npmConfig: ConfigType = {
-	installCmd: "install",
-	dependency: "--save",
-	devDependency: "--save-dev",
-	optionalDependency: "--save-optional"
-};
+interface PackageManagerConfig {
+	[key: string]: Commands;
+}
 
-const yarnConfig: ConfigType = {
-	installCmd: "add",
-	dependency: " ",
-	devDependency: "--save",
-	optionalDependency: "--optional"
+const pmConfig: PackageManagerConfig = {
+	npm: {
+		dependency: ["install", "--save"],
+		devDependency: ["install", "--save-dev"],
+		optionalDependency: ["install", "--save-optional"]
+	},
+	yarn: {
+		dependency: ["add"],
+		devDependency: ["add", "-D"],
+		optionalDependency: ["add", "--optional"]
+	}
 };
 
 const spawnWithArg = (pm: string, cmd: string): SpawnSyncReturns<Buffer> => {
-	const pmConfig: ConfigType = pm === "npm" ? npmConfig : yarnConfig;
-	const options: string[] = [pmConfig.installCmd, "webpack-dev-server", pmConfig[cmd]];
+	/*
```

Comment:
You can move this to a function doc if you see convention used a lil bit further up